src/components/cart/utils.ts

```diff
@@ -0,0 +1,181 @@
+import { useFiatCurrency } from 'hooks/useFiatCurrency';
+import { useTokens } from 'hooks/useTokens';
+import {
+  CartStrategy,
+  CreateStrategyOrder,
+  CreateStrategyParams,
+} from 'libs/queries';
+import { useGetMultipleTokenPrices } from 'libs/queries/extApi/tokenPrice';
+import { SafeDecimal } from 'libs/safedecimal';
+import { useEffect, useMemo, useState } from 'react';
+import { lsService } from 'services/localeStorage';
+import { useWagmi } from 'libs/wagmi';
+import strategyStyle from 'components/strategies/overview/StrategyContent.module.css';
+import formStyle from 'components/strategies/common/form.module.css';
+
+export type Cart = (CreateStrategyParams & { id: string })[];
+
+const toOrder = (sdkOrder: CreateStrategyOrder) => ({
+  balance: sdkOrder.budget,
+  startRate: sdkOrder.min,
+  endRate: sdkOrder.max,
+  marginalRate: sdkOrder.marginalPrice,
+});
+
+export const useStrategyCart = () => {
+  const [cart, setCart] = useState<Cart>([]);
+  const { user } = useWagmi();
+  const { getTokenById } = useTokens();
+  const { selectedFiatCurrency } = useFiatCurrency();
+
+  const tokens = cart.map(({ base, quote }) => [base, quote]).flat();
+  const addresses = Array.from(new Set(tokens));
+  const priceQueries = useGetMultipleTokenPrices(addresses);
+
+  useEffect(() => {
+    if (!user) return setCart([]);
+    const carts = lsService.getItem('carts') ?? {};
+    setCart(carts[user] ?? []);
+  }, [user]);
+
+  useEffect(() => {
+    const handler = (event: StorageEvent) => {
+      if (event.key !== lsService.keyFormatter('carts')) return;
+      if (!user) return;
+      const next = JSON.parse(event.newValue ?? '{}');
+      setCart(next[user] ?? []);
+    };
+    window.addEventListener('storage', handler);
```

Comment:
nitpick but I am not a fan of calling `window.addEventListener('storage', handler);` every place where you care about whether the user changed - and it's already in two places here. We can consider alternatives. A Custom hook will allow reuse and help sticking with a single event handler across the system. Maybe there's a better idea.